pyproject.toml

```diff
@@ -1,9 +1,28 @@
 [tool.ruff]
 exclude = ["main_window.py", "home_ui.py", "resources.py", "resources_rc.py"]
+line-length = 100
+
+[tool.ruff.lint.per-file-ignores]
+"tagstudio/tests/**" = ["D"]
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
 
 [tool.ruff.lint]
-select = ["E", "F", "UP", "B", 'SIM']
-ignore = ["E402", "E501", "F541"]
+select = ["E", "F", "D", "I", "UP", "B", "SIM"]
```

Comment:
should be added:
- `FBT` https://docs.astral.sh/ruff/rules/#flake8-boolean-trap-fbt
- `CPY` https://docs.astral.sh/ruff/rules/#flake8-copyright-cpy
- `T20` https://docs.astral.sh/ruff/rules/#flake8-print-t20

nice to have:
- `SIM` to catch unnecessarily complex code https://docs.astral.sh/ruff/rules/#flake8-simplify-sim


nice to have but will need some more work:
- `N` https://docs.astral.sh/ruff/rules/#pep8-naming-n 
- `AN001` although I'm not sure if that would be possible to apply for new code, as there is a lot of annotations missing in existing code  https://docs.astral.sh/ruff/rules/#flake8-annotations-ann

Reply:
- `FBT` seems well-intentioned but would require a *significant* amount of  additional refactoring work that's I'm not prepared for
- `CPY` seems good, looks like I just need to do some configuring work beforehand
- `T20` looks great
- `N` I like a lot, but will require a whole lot of `noqa: N802` comments for overriding native Qt methods that insist on sticking with camelCase. But I don't mind doing this and reformatting non-Qt methods that may mistakenly be using the wrong case
- `ANN001` I agree with you on, it would be great to have but adds on a lot of work to catch up the existing codebase

Reply:
I'll do `FBT` in a separate PR

Reply:
looks like only `FBT003` is really feasiable